### ISSUE TITLE
MySQL on Yosemite fails to start, tries to use socket: `/socket`

### ISSUE THREAD

**Issue by thisisbrians** [boxen/puppet-mysql#49]
I recently upgraded to Yosemite and I've had trouble getting MySQL up and running through Boxen. I believe I am using the latest version of `boxen/our-boxen`.

Output from `$ boxen --env`:
```
No receipt for 'com.apple.pkg.CLTools_Executables' found at '/'.
BOXEN_BIN_DIR=/opt/boxen/bin
BOXEN_CONFIG_DIR=/opt/boxen/config
BOXEN_DATA_DIR=/opt/boxen/data
BOXEN_ENV_DIR=/opt/boxen/env.d
BOXEN_GITHUB_LOGIN=thisisbrians
BOXEN_HOME=/opt/boxen
BOXEN_LOG_DIR=/opt/boxen/log
BOXEN_MYSQL_HOST=127.0.0.1
BOXEN_MYSQL_PORT=13306
BOXEN_MYSQL_SOCKET=/socket
BOXEN_MYSQL_URL=mysql://thisisbrians@127.0.0.1:13306/
BOXEN_REDIS_HOST=127.0.0.1
BOXEN_REDIS_PORT=16379
BOXEN_REDIS_URL=redis://127.0.0.1:16379/
BOXEN_SETUP_VERSION=7afc997
BOXEN_SOCKET_DIR=/opt/boxen/data/project-sockets
BOXEN_SRC_DIR=/Users/thisisbrians/src
```

My project manifest:
```puppet
class projects::bractlet-beta {

  $ruby_version = '1.9.3-p448'

  boxen::project { 'bractlet-beta':
    ruby   => $ruby_version,
    mysql  => true,
    source => 'Bractlet/bractlet-beta',
    nodejs => 'v0.10.13',
    redis  => true
  }

  # configure redis as a cache according to http://redis.io/topics/config:
  file_line { 'add maxmemory to redis config':
    path    => "${boxen::config::configdir}/redis/redis.conf",
    line    => "maxmemory 2mb",
    notify  =>  Service["redis"]
  }

  file_line { 'add maxmemory policy to redis config':
    path    => "${boxen::config::configdir}/redis/redis.conf",
    line    => "maxmemory-policy allkeys-lru",
    notify  => Service["redis"]
  }

  ruby_gem { "bundler for ruby ${ruby_version}":
    gem  => 'bundler',
    ruby_version => $ruby_version
  }

  ruby_gem { "mailcatcher for ruby ${ruby_version}":
    gem  => 'mailcatcher',
    ruby_version => $ruby_version
  }

}
```

Relevant output from running `$ boxen bractlet-beta`:

```
Notice: /Stage[main]/Mysql::Service/Service[dev.mysql]/ensure: ensure changed 'stopped' to 'running'
Notice: /Stage[main]/Mysql::Setup/Exec[wait-for-mysql]: Triggered 'refresh' from 1 events
Notice: /Stage[main]/Mysql::Setup/Exec[mysql-tzinfo-to-sql]/returns: Warning: Unable to load '/usr/share/zoneinfo/+VERSION' as time zone. Skipping it.
Notice: /Stage[main]/Mysql::Setup/Exec[mysql-tzinfo-to-sql]/returns: ERROR 2003 (HY000): Can't connect to MySQL server on '127.0.0.1' (61)
Error: /Stage[main]/Mysql::Setup/Exec[mysql-tzinfo-to-sql]: Failed to call refresh: mysql_tzinfo_to_sql /usr/share/zoneinfo |       mysql -u root mysql -h127.0.0.1 -P13306 -S/socket returned 1 instead of one of [0]
Error: /Stage[main]/Mysql::Setup/Exec[mysql-tzinfo-to-sql]: mysql_tzinfo_to_sql /usr/share/zoneinfo |       mysql -u root mysql -h127.0.0.1 -P13306 -S/socket returned 1 instead of one of [0]
Notice: /Stage[main]/Mysql::Setup/Exec[grant root user privileges]/returns: ERROR 2003 (HY000): Can't connect to MySQL server on '127.0.0.1' (61)
Error: /Stage[main]/Mysql::Setup/Exec[grant root user privileges]: Failed to call refresh: mysql -uroot --password='' -h127.0.0.1 -P13306 -S/socket       -e 'grant all privileges on *.* to 'root'@'localhost'' returned 1 instead of one of [0]
Error: /Stage[main]/Mysql::Setup/Exec[grant root user privileges]: mysql -uroot --password='' -h127.0.0.1 -P13306 -S/socket       -e 'grant all privileges on *.* to 'root'@'localhost'' returned 1 instead of one of [0]
Notice: /Stage[main]/Projects::Bractlet-beta/Boxen::Project[bractlet-beta]/Mysql::Db[bractlet-beta_development]/Exec[create mysql db bractlet-beta_development]/returns: mysqladmin: connect to server at '127.0.0.1' failed
Notice: /Stage[main]/Projects::Bractlet-beta/Boxen::Project[bractlet-beta]/Mysql::Db[bractlet-beta_development]/Exec[create mysql db bractlet-beta_development]/returns: error: 'Can't connect to MySQL server on '127.0.0.1' (61)'
Notice: /Stage[main]/Projects::Bractlet-beta/Boxen::Project[bractlet-beta]/Mysql::Db[bractlet-beta_development]/Exec[create mysql db bractlet-beta_development]/returns: Check that mysqld is running on 127.0.0.1 and that the port is 13306.
Notice: /Stage[main]/Projects::Bractlet-beta/Boxen::Project[bractlet-beta]/Mysql::Db[bractlet-beta_development]/Exec[create mysql db bractlet-beta_development]/returns: You can check this by doing 'telnet 127.0.0.1 13306'
Error: mysqladmin -h127.0.0.1 -uroot -p13306 create bractlet-beta_development --password='' returned 1 instead of one of [0]
Error: /Stage[main]/Projects::Bractlet-beta/Boxen::Project[bractlet-beta]/Mysql::Db[bractlet-beta_development]/Exec[create mysql db bractlet-beta_development]/returns: change from notrun to 0 failed: mysqladmin -h127.0.0.1 -uroot -p13306 create bractlet-beta_development --password='' returned 1 instead of one of [0]
Notice: /Stage[main]/Projects::Bractlet-beta/Boxen::Project[bractlet-beta]/Mysql::Db[bractlet-beta_test]/Exec[create mysql db bractlet-beta_test]/returns: mysqladmin: connect to server at '127.0.0.1' failed
Notice: /Stage[main]/Projects::Bractlet-beta/Boxen::Project[bractlet-beta]/Mysql::Db[bractlet-beta_test]/Exec[create mysql db bractlet-beta_test]/returns: error: 'Can't connect to MySQL server on '127.0.0.1' (61)'
Notice: /Stage[main]/Projects::Bractlet-beta/Boxen::Project[bractlet-beta]/Mysql::Db[bractlet-beta_test]/Exec[create mysql db bractlet-beta_test]/returns: Check that mysqld is running on 127.0.0.1 and that the port is 13306.
Notice: /Stage[main]/Projects::Bractlet-beta/Boxen::Project[bractlet-beta]/Mysql::Db[bractlet-beta_test]/Exec[create mysql db bractlet-beta_test]/returns: You can check this by doing 'telnet 127.0.0.1 13306'
Error: mysqladmin -h127.0.0.1 -uroot -p13306 create bractlet-beta_test --password='' returned 1 instead of one of [0]
Error: /Stage[main]/Projects::Bractlet-beta/Boxen::Project[bractlet-beta]/Mysql::Db[bractlet-beta_test]/Exec[create mysql db bractlet-beta_test]/returns: change from notrun to 0 failed: mysqladmin -h127.0.0.1 -uroot -p13306 create bractlet-beta_test --password='' returned 1 instead of one of [0]
```

And from `/opt/boxen/log/mysql/error.log` (this just loops incessantly):
```
141029 13:08:48 mysqld_safe Starting mysqld daemon with databases from /opt/boxen/data/mysql
141029 13:08:48 [Warning] Setting lower_case_table_names=2 because file system for /opt/boxen/data/mysql/ is case insensitive
141029 13:08:48 InnoDB: The InnoDB memory heap is disabled
141029 13:08:48 InnoDB: Mutexes and rw_locks use GCC atomic builtins
141029 13:08:48 InnoDB: Compressed tables use zlib 1.2.5
141029 13:08:48 InnoDB: Initializing buffer pool, size = 128.0M
141029 13:08:48 InnoDB: Completed initialization of buffer pool
141029 13:08:48 InnoDB: highest supported file format is Barracuda.
141029 13:08:48  InnoDB: Waiting for the background threads to start
141029 13:08:49 InnoDB: 1.1.8 started; log sequence number 1595675
141029 13:08:49 [ERROR] Can't start server : Bind on unix socket: Permission denied
141029 13:08:49 [ERROR] Do you already have another mysqld server running on socket: /socket ?
141029 13:08:49 [ERROR] Aborting

141029 13:08:49  InnoDB: Starting shutdown...
141029 13:08:49  InnoDB: Shutdown completed; log sequence number 1595675
141029 13:08:49 [Note] /opt/boxen/homebrew/Cellar/mysql/5.5.20-boxen2/bin/mysqld: Shutdown complete

141029 13:08:49 mysqld_safe mysqld from pid file /opt/boxen/data/mysql/Brians-MacBook-Pro.local.pid ended
```

The `/socket` bit definitely doesn't seem right, so I took a look at `/opt/boxen/config/mysql/my.cnf`:
```
[client]
host=127.0.0.1
port=13306
socket=/socket

[server]
datadir=/opt/boxen/data/mysql
log-error=/opt/boxen/log/mysql/error.log
bind-address=127.0.0.1
port=13306
socket=/socket
max_allowed_packet=1073741824
```

I realized that `$BOXEN_MYSQL_SOCKET` in my env is also `/socket`, and I think I've traced the origin of `/socket` back to this repo: https://github.com/boxen/puppet-mysql/blob/master/manifests/config.pp#L17

So, it seems that `${datadir}` is empty, and that's as far as I've gotten with digging on this one. Any ideas on what is going wrong here? Am I barking up the right tree?

**Comment by jesseclark** [comment]
@tob1k Which yaml file did you add that config setting to? 

I tried adding it to ```our_boxen/hiera/common.yaml``` but it didn't have any effect. And when I modified ```our_boxen/shared/mysql/data/Darwin.yaml``` directly it seems to get reverted when I run ```./script/boxen```.

If I copy the whole module from /shared to /modules and modify it there it works but I would prefer not to have to do this manually every time...

**Comment by tob1k** [comment]
@jesseclark Yep- just adding it to `hiera/common.yaml` will do the trick. You just need to make your you commit your change before running `boxen` again, or like you say it'll be reverted.

**Comment by mikemcquaid** [comment]
Looks like this may have been fixed by #52?

**Comment by seanknox** [comment]
@thisisbrians I believe #52 fixes this exact issue; feel free to reopen if you're still having issue.
